cpp/src/join/hash_join.cu

```diff
@@ -196,13 +196,18 @@ get_left_join_indices_complement(rmm::device_vector<size_type> &right_indices,
  * @throw std::out_of_range if elements of `build_on` exceed the number of columns in the `build`
  * table.
  *
- * @param build_table Table of build side columns to join.
+ * @param build_table Device view of table of columns used to build join hash.
+ * @param build Table of columns used to build join hash.
+ * @param compare_nulls Controls whether null join-key values should match or not.
  * @param stream CUDA stream used for device memory operations and kernel launches.
  *
  * @return Built hash table.
  */
 std::unique_ptr<multimap_type, std::function<void(multimap_type *)>> build_join_hash_table(
-  cudf::table_device_view build_table, rmm::cuda_stream_view stream)
+  cudf::table_device_view build_table,
+  cudf::table_view const &build,
```

Comment:
I'm confused. Why was this parameter added?

Reply:
Are `build` and `build_table` the same table?

Reply:
So `build` is the table view that comes in to the `hash_join_impl` constructor and `build_table` is `auto build_table = cudf::table_device_view::create(_build_selected, stream);` which is built from the selected columns. If the selected columns are not all the columns, is it incorrect to use build to check for nulls? I think it is invalid because there could be a column with nulls in it that are not selected.

I wonder if this should just receive the table with the selected columns and create the device portion inside `build_join_hash_table`.

Reply:
> If the selected columns are not all the columns, is it incorrect to use build to check for nulls? I think it is invalid because there could be a column with nulls in it that are not selected.

The "on" columns are the only thing that matter if they contain nulls. 

> I wonder if this should just receive the table with the selected columns and create the device portion inside build_join_hash_table.

That's what I was thinking. Is the `table_device_view` passed into `build_join_hash_table` reused elsewhere?

Reply:
No, created and only used here. I'll work that up.

Reply:
Updated with this request. Thanks for the suggestion and for pushing on what actually was being used here because it uncovered a bug.